### DISCUSSION TITLE
Two New Parameters for the Color class

### DISCUSSION THREAD

**Discussion by MrGibbage** [pybricks/support#1183] [Ideas]
Using a Spike Prime hub. If I create a new Color, say limeGreen = Color(h=92, s=59, v=90), then add that to the detectableColors, the sensor is then quite capable of detecting that color. But if I have a program that detects the color sensed and then prints what it sees, you get the HSV values, and not the informative "limeGreen"
```python
# Color(h=92, s=59, v=90) has been added to detectableColors
colHsv = colorSensor.hsv(True)
print(colorSensor.color() + ": " + str(colHsv))
# This will print 
# Color(h=92, s=59, v=90): Color(h=93, s=57, v=91)

This would be more legible if it looked like
print(colorSensor.color().name + ": " + str(colHsv))
# which would print
# limeGreen: Color(h=93, s=57, v=91)

```

If the Color class had a name parameter, we could set that parameter to the friendly name, such as "limeGreen". Ideally this would be available in the constructor so we could use limeGreen = Color(h=92, s=59, v=90, name="limeGreen"), but of course it could be accessed via the .name parameter as well. Each of the default colors would have these values already populated. We could then access the name of the detected color via colorSensor.color().name.

Secondly, assume that one of the default colors is reconfigured to match more closely to what the color sensor is seeing. For example, my colorSensor sees green as something close to Color.GREEN = Color(h=154, s=69, v=55). But the default value is Color(h=120, s=100, v=100). No problem, I can overwrite the default values with mine and the colorSensor is then much more accurate with detecting green. But, if I try to set the hub light to Color.GREEN afterwards, the light turns to a pale not-green color, due to the low saturation value. I am proposing that the Color class could also have a parameter for "originalColor" (and I'll be honest, I'm not in love with this idea). The originalColor in this case could be set to Color.GREEN's HSV values before being overwritten. Then they would be available for setting the hub light. I have done this in my program, but I just keep track of the colors in a dictionary:

```python
myGreen = Color(h=154, s=69, v=55)
myColor2DefaultColorDict = {
  myGreen : Color.GREEN,
  ... and others)
}
colorSensor.detectableColors([myGreen, and others])
```

I'm not in love with adding the originalColor to the Color class, namely because I think this is the only time it would be needed (using an overwritten color to try and set the hub light color). I could also write my own MyColor class and add the original color information, so this is not a big deal. Same for the "name" information, but I think many more people would find the name parameter useful.

**Comment by dlech** [comment]
If you don't change the value of the "constants", then we don't need the originalColor paramter. You can also add your own colors to the `Color` dict to provide a name for them so we can probably live without a name parameter as well.

```python
>>> Color.DARK_GREEN = Color(120, 50, 50)
>>> Color.DARK_GREEN
Color.DARK_GREEN
```

**Reply by MrGibbage** [reply]
Understood. Thanks for the feedback. That is basically what I ended up doing.
